### DISCUSSION TITLE
Supabase session string storage

### DISCUSSION THREAD

**Discussion by brycegoh** [supabase/supabase#2061] [Questions] [answered]
Hello. Would like to ask what are the best practices and methods used to store the session string for persistent login on a mobile app developed using flutter.

Currently, I am using flutter secure storage to store the session string. However, when the user uninstalls the app, the session string does not get deleted because flutter secure storage stores in IOS keychain / android keystore.

Therefore, how should I approach storing the session string?

Thanks!

**Comment by phamhieu** [comment, marked as answer]
Hi @brycegoh, you can take a look at Supabase flutter example app [here](https://github.com/phamhieu/supabase-dart-demo). It uses `shared_preferences` package to persist the session string.

As the session string will be renewed and updated frequently (default in 1 hour). You don't need to use flutter secure storage.

**Reply by sumchans** [reply]
@phamhieu Hello, Does this solution work as of now? I am also trying to preserve the session status after the user signs up with phone in flutter. I have heard there is an issue with gotrue not returning response data.

**Reply by phamhieu** [reply]
@sumchans we released a new flutter package for Supabase recently. You can take a look https://pub.dev/packages/supabase_flutter. It supports secure storage for session string.

> I have heard there is an issue with gotrue not returning response data.

Supabase is on Discord now http://discord.supabase.com. You are welcome to join the flutter group chat.
Moreover If you need help from the team, please email beta@supabase.io or create a github issue on related repos. Thank you.

**Reply by sumchans** [reply]
@phamhieu Wow didn't know about that. I got the package updated now. I tried with the new update but still I am getting directed to the login page even if the user has signed in before. Are there any samples that I could follow using the new update. I am currently following the supabase flutter quickstart tutorial. Every single time I run the app I see the message on the console saying Current session is missing data.

**Reply by phamhieu** [reply]
@sumchans can you create an issue on https://github.com/supabase/supabase-flutter with the error details.

**Reply by sumchans** [reply]
@phamhieu  Just created an issue.
